internal/k8s/epslices/endpoint_slices.go

```diff
@@ -11,16 +11,13 @@ import (
 
 const SlicesServiceIndexName = "ServiceName"
 
-// IsConditionServing tells if the conditions represent a serving state, deferring
-// to ready state if serving == nil.
-func IsConditionServing(conditions discovery.EndpointConditions) bool {
-	if conditions.Serving == nil {
-		if conditions.Ready == nil {
-			return true
-		}
-		return *conditions.Ready
+// IsConditionReady tells if the conditions represent a ready state, deferring
+// to serving state if ready == false.
+func IsConditionReady(conditions discovery.EndpointConditions) bool {
+	if !(conditions.Ready == nil || *conditions.Ready) {
```

Comment:
I am having a hard time parsing this. Can you unroll the if and have each exit point separate (or just take inspiration to what cilium does?)